protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -1235,7 +1232,7 @@ impl ProxyExtendedChannelFactory {
                     Some(custom_mining_job),
                 ));
             }
-            if new_job.future_job {
+            if new_job.is_future() {
```

Comment:
converting to `else if` would be a little more efficient

Reply:
changed first condition exclude the second one so we can safely use else if